### ISSUE TITLE
[Feature Request] Pull/Fetch all (Open Tabs/Folder/Repositories)

### ISSUE THREAD

**Issue by Goepfie** [ForkIssues/TrackerWin#2140]
Use Case:
I have a organized my repositories in the manager with neat little folders. Every time I switch my device (Desktop to Laptop for example). I need to manually fetch & pull all of my repos single-handedly. Especially my Framework, which consists of 20+ repositories which get symbolically linked into my dev environment are a pain to maintain, especially when they do not have local changes.

Need:
I would want a command from the UI which iterates into every repo open in the tab/existent in the workspace/selected in the repo manager and fast forwards every one in it's current branch, if possible. I would not mind having new tabs pop up, for repos which produce an error. It would be preferably to have each process start in parallel than in sequence. A shared progress bar would be nice to have.

I am not sure how this would be possible with a custom command, mainly because there is no option for attaching a script in the necessary top level selections. The option of a command line script produces a lot of hurdles for me, especially regarding key-management and defeat the purpose of having Fork for me.

Thanks for hearing me out and Fork is Awesome! Great work. So much better than most of the products done by bigger enterprise teams.

**Comment by DanPristupov** [comment]
Regarding fetch:

Open all your repos as tabs and they will be fetched every 5 minutes.

Regarding pull:

Pull is a merge and automatic merge of multiple repos will likely never appear because it's error prone.


> Especially my Framework, which consists of 20+ repositories which get symbolically linked into my dev environment are a pain to maintain, especially when they do not have local changes.

Can submodules be a solution for you? It will fetch and checkout corresponding commits for you (submodules will also bring additional problems of course 🙂)

**Comment by Shineson1001** [comment]
Hi,
maybe like in firefox. You can select all or some tabs and do actions on the selected tabs like "Reload Tabs" ...
![grafik](https://github.com/fork-dev/TrackerWin/assets/22273771/5a267c46-0615-4578-b092-dd2c212aa74b)
